### PR TITLE
Origin/charity/admin

### PR DESCRIPTION
**Pull request by zhaulanbekk**

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import kg.peaksoft.giftlistb6.dto.responses.CharityResponses;
+import kg.peaksoft.giftlistb6.dto.responses.InnerCharityResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/charities")
+@RequiredArgsConstructor
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Admin charity Api", description = "")
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java]: write description pls

File: src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import io.swagger.v3.oas.annotations.tags.Tag;
+import kg.peaksoft.giftlistb6.db.services.AdminCharityService;
+import kg.peaksoft.giftlistb6.dto.responses.CharityResponses;
+import kg.peaksoft.giftlistb6.dto.responses.InnerCharityResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/charities")
+@RequiredArgsConstructor
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java]: write PreAuthorize

File: src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Admin charity Api", description = "")
+public class AdminCharityApi {
+
+    private final AdminCharityService charityService;
+
+    @Operation(summary = "Get charity", description = "Get charity by id")
+    @GetMapping("{id}/admin")
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java]: delete admin

File: src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java [hunk 4]
```diff
@@ -38,6 +38,20 @@ public interface CharityRepository extends JpaRepository<Charity, Long> {
...
+            "ch.id," +
+            "ch.image," +
+            "ch.name," +
+            "ch.condition," +
+            "ch.createdDate," +
+            "ch.charityStatus," +
+            "ch.user.id," +
+            "ch.user.firstName," +
+            "ch.user.lastName," +
+            "ch.user.photo )"+
```
Comment (maksat990rd) [inline src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java]: rename photo on image

Comment (zhaulanbekk) [reply src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java]: men jazgan emesmin bul entitide ushintip jazylyptyr

File: src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/admin/charities")
+@RequiredArgsConstructor
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Admin charity Api", description = "Admin can see all charities ")
+@PreAuthorize("hasAuthority('ADMIN')")
+public class AdminCharityApi {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/AdminCharityApi.java]: Delete this class and move those endpoints to AdminApi

File: src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java (hunk 4 above)
Comment (i-Coderr) [reply src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java]: Сиз оңдоп койуңуз image деп

File: src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java [hunk 6]
```diff
@@ -38,6 +38,20 @@ public interface CharityRepository extends JpaRepository<Charity, Long> {
             "from User u join u.charities ch where u.email <> ?1")
     List<OtherCharityResponse> getAllOther(String email);
 
+    @Query("select new kg.peaksoft.giftlistb6.dto.responses.OtherCharityResponse("+
+            "ch.id," +
+            "ch.image," +
+            "ch.name," +
+            "ch.condition," +
+            "ch.createdDate," +
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/repositories/CharityRepository.java]: Rename to createdAt

**Review by i-Coderr — COMMENTED** [review]
Commented at 8.28 pm

File: src/main/java/kg/peaksoft/giftlistb6/db/services/AdminCharityService.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import kg.peaksoft.giftlistb6.exceptions.NotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class AdminCharityService {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/AdminCharityService.java]: Delete this class and move those methods to CharityService class

**Review by i-Coderr — COMMENTED** [review]
Commented at 9.08 pm

File: src/main/java/kg/peaksoft/giftlistb6/api/AdminApi.java [hunk 8]
```diff
@@ -38,4 +40,16 @@ public SimpleResponse block(@PathVariable("id") Long id) {
...
     }
+
+    @Operation(summary = "Get charity", description = "Get charity by id")
+    @GetMapping("{id}")
+    public InnerCharityResponse getCharityWithId(@PathVariable Long id) {
+        return adminService.getCharityById(id);
+    }
+
+    @Operation(summary = "All charities", description = "Admin can see all charities")
+    @GetMapping
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/api/AdminApi.java]: add ("charities")

**Review by i-Coderr — COMMENTED** [review]
Commented at 8.37 pm

**Comment by i-Coderr** [comment]
Pls resolve conflicts

**Pull request merged**